app/controllers/api/base_controller/parameters.rb

```diff
@@ -160,20 +160,24 @@ def sort_params(klass)
         orders = String(params['sort_order']).split(",")
         options = String(params['sort_options']).split(",")
         params['sort_by'].split(",").zip(orders).collect do |attr, order|
-          if klass.attribute_method?(attr) || klass.method_defined?(attr) || attr == klass.primary_key
-            sort_directive(attr, order, options)
+          if klass.virtual_attribute?(attr) && !klass.attribute_supported_by_sql?(attr)
```

Comment:
While the simpler path is more elegant, I like what @jntullo did to have the 2 different error messages.   I'm ok with this approach here as this is low count iteration.